core/shared/src/main/scala/org/typelevel/log4cats/SamStructuredLogger.scala

```diff
@@ -0,0 +1,266 @@
+    new SamStructuredLogger[F] {
+      protected def kernel: LoggerKernel[F, String] = new LoggerKernel[F, String] {
+        def log(
+            level: KernelLogLevel,
+            record: Log.Builder[String] => Log.Builder[String]
+        ): F[Unit] =
+          f(logger.kernel.log(level, record))
+      }
+
+      override def trace(ctx: Map[String, String])(msg: => String): F[Unit] =
+        f(logger.trace(ctx)(msg))
+      override def debug(ctx: Map[String, String])(msg: => String): F[Unit] =
+        f(logger.debug(ctx)(msg))
+      override def info(ctx: Map[String, String])(msg: => String): F[Unit] =
+        f(logger.info(ctx)(msg))
+      override def warn(ctx: Map[String, String])(msg: => String): F[Unit] =
+        f(logger.warn(ctx)(msg))
+      override def error(ctx: Map[String, String])(msg: => String): F[Unit] =
+        f(logger.error(ctx)(msg))
+
+      override def trace(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+        f(logger.trace(ctx, t)(msg))
+      override def debug(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+        f(logger.debug(ctx, t)(msg))
+      override def info(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+        f(logger.info(ctx, t)(msg))
+      override def warn(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+        f(logger.warn(ctx, t)(msg))
+      override def error(ctx: Map[String, String], t: Throwable)(msg: => String): F[Unit] =
+        f(logger.error(ctx, t)(msg))
+
+      override def trace(message: => String): F[Unit] = f(logger.trace(message))
+      override def trace(t: Throwable)(message: => String): F[Unit] = f(logger.trace(t)(message))
+      override def debug(message: => String): F[Unit] = f(logger.debug(message))
+      override def debug(t: Throwable)(message: => String): F[Unit] = f(logger.debug(t)(message))
+      override def info(message: => String): F[Unit] = f(logger.info(message))
+      override def info(t: Throwable)(message: => String): F[Unit] = f(logger.info(t)(message))
+      override def warn(message: => String): F[Unit] = f(logger.warn(message))
+      override def warn(t: Throwable)(message: => String): F[Unit] = f(logger.warn(t)(message))
+      override def error(message: => String): F[Unit] = f(logger.error(message))
+      override def error(t: Throwable)(message: => String): F[Unit] = f(logger.error(t)(message))
+    }
+
+  implicit def optionTSamStructuredLogger[F[_]: Functor](implicit
+      ev: SamStructuredLogger[F]
+  ): SamStructuredLogger[OptionT[F, *]] =
+    ev.mapK(OptionT.liftK[F])
+
+  implicit def eitherTSamStructuredLogger[F[_]: Functor, E](implicit
+      ev: SamStructuredLogger[F]
+  ): SamStructuredLogger[EitherT[F, E, *]] =
+    ev.mapK(EitherT.liftK[F, E])
+
+  implicit def kleisliSamStructuredLogger[F[_], A](implicit
+      ev: SamStructuredLogger[F]
+  ): SamStructuredLogger[Kleisli[F, A, *]] =
+    ev.mapK(Kleisli.liftK[F, A])
```

Comment:
We _really_ don't want to add implicit conversions for this sort of thing. It leads to really confusing code.